c7n/resources/ec2.py

```diff
@@ -1547,18 +1547,22 @@ def process(self, instances):
 
     def disable_deletion_protection(self, client, instances):
 
-        def process_instance(i):
+        def modify_instance(i, attribute):
             try:
                 self.manager.retry(
                     client.modify_instance_attribute,
                     InstanceId=i['InstanceId'],
-                    Attribute='disableApiTermination',
+                    Attribute=attribute,
                     Value='false')
             except ClientError as e:
                 if e.response['Error']['Code'] == 'IncorrectInstanceState':
                     return
                 raise
 
+        def process_instance(i):
+            modify_instance(i, 'disableApiTermination')
+            modify_instance(i, 'disableApiStop')
```

Comment:
I like how you factored the call out to make it clearer that we need to set 2 attributes. One edge case though: if this policy fires in an environment with an older botocore (prior to [1.26.7](https://github.com/boto/botocore/commit/4537c0fbf874c5020567969f975c1091b08d7ab4#diff-2c623f3c6a917be56c59d43279244996836262cb1e12d9d0786c9c49eef6b43cR12)) we'll error out trying to update the attribute. We might be able to fall back to the original behavior in that case (handle termination protection only) but that's not particularly clear without a visible warning and/or some notes in the docs.

Reply:
Based on some local testing with an older versions of boto3/botocore[1], this API call should still work:
```
import boto3; 
ec2 = boto3.client('ec2', region_name='us-west-2')
# Does work
ec2.modify_instance_attribute(InstanceId="i-something", Attribute="disableApiStop", Value="false")


# Does not work
ec2.modify_instance_attribute(InstanceId="i-something", DisableApiStop={"Value" : False})
```

[1]
```
boto3                                    1.20.0
botocore                                 1.23.54
```

Reply:
I'm not sure if that's representative of how Custodian works. I can definitely catch this shape of issue (`ParamValidationError`) & add docs/warning if that happens

Reply:
Even with the new botocore versions, this would need to be two requests 😢 
```
In [11]: ec2.modify_instance_attribute(InstanceId=me, DisableApiStop={"Value" : False}, DisableApiTermination={"Value": False})
---------------------------------------------------------------------------
ClientError                               Traceback (most recent call last)
<ipython-input-11-3946a3a56f62> in <module>
----> 1 ec2.modify_instance_attribute(InstanceId=me, DisableApiStop={"Value" : False}, DisableApiTermination={"Value": False})

~/anaconda3/lib/python3.8/site-packages/botocore/client.py in _api_call(self, *args, **kwargs)
    506                 )
    507             # The "self" in this scope is referring to the BaseClient.
--> 508             return self._make_api_call(operation_name, kwargs)
    509 
    510         _api_call.__name__ = str(py_operation_name)

~/anaconda3/lib/python3.8/site-packages/botocore/client.py in _make_api_call(self, operation_name, api_params)
    913             error_code = parsed_response.get("Error", {}).get("Code")
    914             error_class = self.exceptions.from_code(error_code)
--> 915             raise error_class(parsed_response, operation_name)
    916         else:
    917             return parsed_response

ClientError: An error occurred (InvalidParameterCombination) when calling the ModifyInstanceAttribute operation: Fields for multiple attribute types specified: disableApiTermination,disableApiStop

```